From badd902ecbda34ac2899494f40d91011a9b34746 Mon Sep 17 00:00:00 2001
From: Alessandro Astone <ales.astone@gmail.com>
Date: Fri, 18 Jul 2025 15:23:29 +0000
Subject: [PATCH 7/8] vold: Disable AbortFuseConnections

This is unsafe to run in a container, because it affects FUSE mounts outside
of the container since fusect has no container namespace isolation.

I'm not yet sure if this has relevant consequences, but the downsides outweigh
any possible issue this introduces in Android.

Change-Id: Ie0acab421b02070e9179b35f2e1aed8c8f51ca3a
---
 Utils.cpp | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/Utils.cpp b/Utils.cpp
index 23630bf..d835555 100644
--- a/Utils.cpp
+++ b/Utils.cpp
@@ -1488,6 +1488,10 @@ bool writeStringToFile(const std::string& payload, const std::string& filename)
 }
 
 status_t AbortFuseConnections() {
+    // Waydroid: this is not safe because fusectl has no container isolation
+    // so we would be aborting host FUSE connections.
+    return OK;
+
     namespace fs = std::filesystem;
 
     static constexpr const char* kFuseConnections = "/sys/fs/fuse/connections";
-- 
2.55.0

